hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -132,11 +132,15 @@ class DefaultSource extends RelationProvider
 
     log.info("Constructing hoodie (as parquet) data source with options :" + optParams)
     // simply return as a regular parquet relation
-    DataSource.apply(
+    val relation =  DataSource.apply(
       sparkSession = sqlContext.sparkSession,
       userSpecifiedSchema = Option(schema),
       className = "parquet",
       options = optParams)
       .resolveRelation()
+
+    sqlContext.sparkContext.hadoopConfiguration.unset("mapreduce.input.pathFilter.class")
```

Comment:
@bvaradar keeping this code actually. given now, we are adding support for MOR snapshot query, if we don't unset this, then if you do a RO query and then a snapshot query, then this will filter out all except latest base files. which will be problematic. 
cc @garyli1019 does this make sense?  Let me see if I can add a test case for this

Reply:
yes make sense. We unset this before the incremental query. When we only have two data source query type, it's fine to unset before running incremental query, but right now we have to unset this here. Still wondering why the local build is able to pass though...